### PR TITLE
Fix json inventory formatter configs

### LINKED ISSUE #918: json inventory filters does not support cfgs
I seem to have overlooked this in #703.  We should also have integration tests that specifically verify the json output

### PR DESCRIPTION
**Pull request by 14rcole**
Fixes #918

### CODE REVIEW

File: linchpin/InventoryFilters/JSONInventoryFormatter.py [hunk 1]
```diff
@@ -85,11 +85,13 @@ def add_common_vars(self, host_groups, layout):
                 self.config["_meta"]["hostvars"] = {}
             for item in items:
                 for var in cvrs:
-                    self.config["_meta"]["hostvars"][item] = {}
-                    if cvrs[var] == "__IP__":
```
Comment (p3ck) [inline linchpin/InventoryFilters/JSONInventoryFormatter.py]: We don't need this special case anymore?

File: linchpin/InventoryFilters/JSONInventoryFormatter.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def add_ips_to_groups(self, inven_hosts, layout):
                         self.config["all"]["hosts"].append(ip)
         return True
 
-    def add_common_vars(self, host_groups, layout):
+    def add_common_vars(self, host_groups, layout, config):
```
Comment (p3ck) [inline linchpin/InventoryFilters/JSONInventoryFormatter.py]: I see a new argument added here but I don't see any references to the code that calls this.

Comment (14rcole) [reply linchpin/InventoryFilters/JSONInventoryFormatter.py]: The code that calls it was updated in #703.  It's the same as the code that calls this method in `CFGInventoryFormatter.py`

File: linchpin/InventoryFilters/JSONInventoryFormatter.py (hunk 1 above)
Comment (14rcole) [reply linchpin/InventoryFilters/JSONInventoryFormatter.py]: No, if `__IP__` isn't defined in the config it's added with a default value, so this case will be handled in the for loop below (`for cfg_item in config:`)

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**